src/applyConstructedStylesPatch.js

```diff
@@ -0,0 +1,52 @@
+// Helper to ensure our custom recorded array exists on the sheet.
+function ensureRecord(sheet) {
+  if (!sheet.__recordedCssRules) {
+    sheet.__recordedCssRules = [];
+  }
+}
+
+module.exports = function applyConstructedStylesPatch() {
+  if (typeof CSSStyleSheet === 'undefined') {
+    console.error('CSSStyleSheet is not supported in this browser');
+    return;
+  }
+  // Patch insertRule to record each rule string.
+  const originalInsertRule = CSSStyleSheet.prototype.insertRule;
+  CSSStyleSheet.prototype.insertRule = function (rule, index) {
+    ensureRecord(this);
+    // If index is not provided or invalid, default to appending.
+    if (index === undefined || index < 0 || index > this.__recordedCssRules.length) {
+      this.__recordedCssRules.push(rule);
+    } else {
+      this.__recordedCssRules.splice(index, 0, rule);
+    }
+    return originalInsertRule.call(this, rule, index);
+  };
+
+  // Patch deleteRule so that removed rules are taken out of our record.
+  const originalDeleteRule = CSSStyleSheet.prototype.deleteRule;
+  CSSStyleSheet.prototype.deleteRule = function (index) {
+    ensureRecord(this);
+    if (index >= 0 && index < this.__recordedCssRules.length) {
+      this.__recordedCssRules.splice(index, 1);
+    }
+    return originalDeleteRule.call(this, index);
+  };
```

Comment:
We should also patch addRule and removeRule in case people are using those.

Reply:
Okay, I'll look into that.